### PR TITLE
Add check for post_types key #33179

### LINKED ISSUE #33179: Argument #2 ($haystack) must be of type array, null given
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When a FSE child theme fails to declare all (both child and parent) template files in its theme.json file -- this can happen when the parent theme is updated with an new template file for example -- trying to edit an existing or create a new Woocommerce products will cause this fatal error:

```
PHP Fatal error:  Uncaught TypeError: in_array(): Argument woocommerce/woocommerce#2 ($haystack) must be of type array, null given in /home/xxx/wp-content/plugins/woocommerce/includes/admin/class-wc-admin-meta-boxes.php:279
Stack trace:
#0 /home/xxx/wp-content/plugins/woocommerce/includes/admin/class-wc-admin-meta-boxes.php(279): in_array()
woocommerce/woocommerce#1 /home/hmkc5789/franceskydive.fr/wp-includes/class-wp-hook.php(309): WC_Admin_Meta_Boxes->remove_block_templates()
woocommerce/woocommerce#2 /home/xxx/wp-includes/plugin.php(189): WP_Hook->apply_filters()
woocommerce/woocommerce#3 /home/xxx/wp-includes/class-wp-theme.php(1334): apply_filters()
woocommerce/woocommerce#4 /home/xxx/wp-admin/includes/theme.php(145): WP_Theme->get_page_templates()
woocommerce/woocommerce#5 /home/xxx/wp-admin/includes/meta-boxes.php(1504): get_page_templates()
woocommerce/woocommerce#6 /home/xxx/wp-admin/edit-form-advanced.php(271): register_and_do_post_meta_boxes()
woocommerce/woocommerce#7 /home/xxx/wp-admin/post-new.php(75): require('/home/xxx/...')
woocommerce/woocommerce#8 {main}
  thrown in /home/hmkc5789/franceskydive.fr/wp-content/plugins/woocommerce/includes/admin/class-wc-admin-meta-boxes.php on line 279
```

### Expected behavior

Being able to edit or create Products.

### Actual behavior

Blank page when trying to edit or create Products.

### Steps to reproduce

Switch to a FSE theme that not has all its template files properly defined in theme.json

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://franceskydive.fr
Site address (URL): https://franceskydive.fr
WC Version: 6.5.1
REST API Version: ✔ 6.5.1
WC Blocks Version: ✔ 7.4.3
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 5.9.3
WP Multisite: –
WP Memory Limit: 256 Mo
WP Debug Mode: –
WP Cron: ✔
Language: fr_FR
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 8.0.19
PHP Post Max Size: 8 Mo
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.71.0
OpenSSL/1.1.1d

SUHOSIN Installed: –
MySQL Version: 10.3.35-MariaDB
Max Upload Size: 8 Mo
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.5.1
WC Database Prefix: wppr_
Taille totale de la base de données: 12.32MB
Taille de la base de données: 9.33MB
Taille de l’index: 2.99MB
wppr_woocommerce_sessions: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_woocommerce_api_keys: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_woocommerce_attribute_taxonomies: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_woocommerce_downloadable_product_permissions: Données : 0.02MB + Index : 0.06MB + Moteur InnoDB
wppr_woocommerce_order_items: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_woocommerce_order_itemmeta: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_woocommerce_tax_rates: Données : 0.02MB + Index : 0.06MB + Moteur InnoDB
wppr_woocommerce_tax_rate_locations: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_woocommerce_shipping_zones: Données : 0.02MB + Index : 0.00MB + Moteur InnoDB
wppr_woocommerce_shipping_zone_locations: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_woocommerce_shipping_zone_methods: Données : 0.02MB + Index : 0.00MB + Moteur InnoDB
wppr_woocommerce_payment_tokens: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_woocommerce_payment_tokenmeta: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_woocommerce_log: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_actionscheduler_actions: Données : 0.05MB + Index : 0.11MB + Moteur InnoDB
wppr_actionscheduler_claims: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_actionscheduler_groups: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_actionscheduler_logs: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_commentmeta: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_comments: Données : 0.02MB + Index : 0.09MB + Moteur InnoDB
wppr_gla_budget_recommendations: Données : 0.22MB + Index : 0.14MB + Moteur InnoDB
wppr_gla_merchant_issues: Données : 0.02MB + Index : 0.00MB + Moteur InnoDB
wppr_gla_shipping_rates: Données : 0.02MB + Index : 0.05MB + Moteur InnoDB
wppr_gla_shipping_times: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_kbp_countdown_entry: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_kbp_form_entry: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_kbp_form_entrymeta: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_links: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_litespeed_url: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_litespeed_url_file: Données : 0.02MB + Index : 0.08MB + Moteur InnoDB
wppr_mailpoet_custom_fields: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_mailpoet_dynamic_segment_filters: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_mailpoet_feature_flags: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_mailpoet_forms: Données : 0.02MB + Index : 0.00MB + Moteur InnoDB
wppr_mailpoet_log: Données : 0.02MB + Index : 0.00MB + Moteur InnoDB
wppr_mailpoet_mapping_to_external_entities: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_mailpoet_newsletters: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_mailpoet_newsletter_links: Données : 0.02MB + Index : 0.05MB + Moteur InnoDB
wppr_mailpoet_newsletter_option: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_mailpoet_newsletter_option_fields: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_mailpoet_newsletter_posts: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_mailpoet_newsletter_segment: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_mailpoet_newsletter_templates: Données : 2.06MB + Index : 0.00MB + Moteur InnoDB
wppr_mailpoet_scheduled_tasks: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_mailpoet_scheduled_task_subscribers: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_mailpoet_segments: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_mailpoet_sending_queues: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_mailpoet_settings: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_mailpoet_statistics_bounces: Données : 0.02MB + Index : 0.00MB + Moteur InnoDB
wppr_mailpoet_statistics_clicks: Données : 0.02MB + Index : 0.05MB + Moteur InnoDB
wppr_mailpoet_statistics_forms: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_mailpoet_statistics_newsletters: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_mailpoet_statistics_opens: Données : 0.02MB + Index : 0.08MB + Moteur InnoDB
wppr_mailpoet_statistics_unsubscribes: Données : 0.02MB + Index : 0.05MB + Moteur InnoDB
wppr_mailpoet_statistics_woocommerce_purchases: Données : 0.02MB + Index : 0.06MB + Moteur InnoDB
wppr_mailpoet_stats_notifications: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_mailpoet_subscribers: Données : 0.02MB + Index : 0.13MB + Moteur InnoDB
wppr_mailpoet_subscriber_custom_field: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_mailpoet_subscriber_ips: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_mailpoet_subscriber_segment: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_mailpoet_user_agents: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_mailpoet_user_flags: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_options: Données : 4.05MB + Index : 0.09MB + Moteur InnoDB
wppr_postmeta: Données : 0.11MB + Index : 0.06MB + Moteur InnoDB
wppr_posts: Données : 1.06MB + Index : 0.06MB + Moteur InnoDB
wppr_termmeta: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_terms: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_term_relationships: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_term_taxonomy: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_tm_taskmeta: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_tm_tasks: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_usermeta: Données : 0.05MB + Index : 0.03MB + Moteur InnoDB
wppr_users: Données : 0.02MB + Index : 0.05MB + Moteur InnoDB
wppr_wc_admin_notes: Données : 0.05MB + Index : 0.00MB + Moteur InnoDB
wppr_wc_admin_note_actions: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_wc_bookings_availability: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_wc_bookings_availabilitymeta: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_wc_booking_relationships: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_wc_category_lookup: Données : 0.02MB + Index : 0.00MB + Moteur InnoDB
wppr_wc_customer_lookup: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_wc_download_log: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_wc_order_coupon_lookup: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_wc_order_product_lookup: Données : 0.02MB + Index : 0.06MB + Moteur InnoDB
wppr_wc_order_stats: Données : 0.02MB + Index : 0.05MB + Moteur InnoDB
wppr_wc_order_tax_lookup: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_wc_product_attributes_lookup: Données : 0.02MB + Index : 0.03MB + Moteur InnoDB
wppr_wc_product_download_directories: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_wc_product_meta_lookup: Données : 0.02MB + Index : 0.09MB + Moteur InnoDB
wppr_wc_rate_limits: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_wc_reserved_stock: Données : 0.02MB + Index : 0.00MB + Moteur InnoDB
wppr_wc_tax_rate_classes: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB
wppr_wc_webhooks: Données : 0.02MB + Index : 0.02MB + Moteur InnoDB

### Post Type Counts ###

attachment: 25
bookable_resource: 3
custom_css: 1
mailpoet_page: 1
page: 9
post: 2
product: 65
product_variation: 7
revision: 3
shop_order: 2
wc_booking: 2
wp_block: 1
wp_global_styles: 3
wp_navigation: 3
wp_template_part: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (7) ###

Kadence Blocks - PRO Extension: par Kadence WP – 1.7.10
Kadence Blocks – Gutenberg Blocks for Page Builder Features: par Kadence WP – 2.4.4
LiteSpeed Cache: par Technologies LiteSpeed – 4.6
Maintenance: par WebFactory Ltd – 4.06
WooCommerce Bookings: par WooCommerce – 1.15.53
WooCommerce Stripe Gateway: par WooCommerce – 6.4.0
WooCommerce: par Automattic – 6.5.1

### Inactive Plugins (6) ###

Advanced WordPress Backgrounds: par nK – 1.8.1
Akismet Anti-Spam: par Automattic – 4.2.4
Jetpack: par Automattic – 10.9.1
MailPoet 3 (New): par MailPoet – 3.89.2
WooCommerce Payments: par Automattic – 4.1.0
WooCommerce Shipping & Tax: par WooCommerce – 1.25.28

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: right_space
Thousand Separator: 
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: booking (booking)
external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: –

### WC Pages ###

Base de la boutique: woocommerce/woocommerce#5 - /boutique/
Panier: woocommerce/woocommerce#6 - /panier/
Commander: woocommerce/woocommerce#7 - /commander/
Mon compte: woocommerce/woocommerce#8 - /mon-compte/
Conditions générales de vente et d’utilisation: ❌ La visibilité de la page doit être publique

### Theme ###

Name: Verry Twenty-Two
Version: 0.1-beta57
Author URL: https://phareo.eu/
Child Theme: ✔
Parent Theme Name: Twenty Twenty-Two
Parent Theme Version: 1.1
Parent Theme Author URL: https://fr.wordpress.org/
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Terminé: 46
Oldest: 2022-04-23 02:24:55 +0200
Newest: 2022-05-23 19:57:13 +0200

En attente: 2
Oldest: 2022-05-24 08:49:39 +0200
Newest: 2022-06-05 14:13:41 +0200

### Status report information ###

Generated at: 2022-05-23 22:46:37 +02:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by gigitux**
This PR adds a check to be sure that we are checking an array. In this way, we can avoid a PHP warning.
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #33179.

### How to test the changes in this Pull Request:

Check out this branch.

1. Install and active `Quadrat` theme.
2. Go on the page for adding a product (`/wp-admin/post-new.php?post_type=product`)
3. Be sure that it is not raised the warning: `Warning: in_array() expects parameter 2 to be array, null given in /plugins/woocommerce/includes/admin/class-wc-admin-meta-boxes.php on line 279`

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm nx changelog <project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33180/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `add changelog` a59a23d4497e5d4d093c3aa6f799e94133bdd8c5
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

**Comment by Aljullu** [comment]
Sorry for the short notice, @vedanshujain, but will you have some spare cycles to review this PR soon? We would like to request a Code Freeze Exception once it's approved if we still have time to get it included in WC core 6.6. There have been some reports about this fatal in the forums and the fix seems very safe and unlikely to cause regressions.

cc @jonathansadowski

File: plugins/woocommerce/changelog/fix-33179-check-type
Comment (barryhughes) [inline plugins/woocommerce/changelog/fix-33179-check-type]: Minor thing, but should this be an actual changelog entry?

Seems like the condition we're fixing has been in place since [6.5.0](https://github.com/woocommerce/woocommerce/blob/6.5.0/plugins/woocommerce/includes/admin/class-wc-admin-meta-boxes.php#L279) so the potential for warnings to be emitted when `$block_template` is null has existed since that point.

**Review by barryhughes — COMMENTED** [review]
Looks good, guards against errors when `$block_template` is null (and if it is a `WP_Error` instance). Left a question regarding changelog, otherwise seems great!

File: plugins/woocommerce/changelog/fix-33179-check-type
Comment (Aljullu) [reply plugins/woocommerce/changelog/fix-33179-check-type]: ~Thanks for the review, @barryhughes. I'm not sure to understand your point, do you suggest we should rephrase the changelog text entry or completely drop it?~

Never mind, @gigitux explained to me that if it has the `Comment: ` prefix, that's not an actual changelog entry, so ignore my previous comment. :sweat_smile:

**Comment by gigitux** [comment]
Thanks for the review!

I updated the PR ([a59a23d](https://github.com/woocommerce/woocommerce/pull/33180/commits/a59a23d4497e5d4d093c3aa6f799e94133bdd8c5)) adding the changelog. Hope that it is fine 👍

File: plugins/woocommerce/changelog/fix-33179-check-type
Comment (barryhughes) [reply plugins/woocommerce/changelog/fix-33179-check-type]: Yes, sorry—could have have been clearer there :+1:

**Review by barryhughes — APPROVED** [review]

**Comment by jonathansadowski** [comment]
This has been cherry-picked into the release branch

**Pull request merged**
